lib/options_validator.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const isObject = require('./utils').isObject;
+
+let validationLevel = 'warn';
+
+function createOptionsSchema(optionsSchema) {
+  return validationFunction.bind(optionsSchema);
+}
+
+function validationFunction(providedOptions) {
+  let validatedOptions = Object.assign({}, providedOptions);
+
+  Object.keys(validatedOptions).forEach(option => {
```

Comment:
Sorry, I accidentally deleted your comment! Here it is: 

- nit: Object.keys(...).forEach used to be much slower than a for loop over the keys, might be worth evaluating the performance difference

They're showing pretty similar performance for me (1.6 ms vs 1.5 ms) and I think `Object.keys(...).forEach` looks much cleaner.

Reply:
What's your benchmark look like? This [jsperf](https://jsperf.com/object-keys-vs-for-in-with-closure/3) indicates its still about 2x faster to do it in a for loop. I'm not one to be a huge stickler here, but you're writing a method that will be applied to every single function in the driver, so it behooves us to consider performance here imo

Reply:
I'll change this to a `for` loop.

My [jsperf](https://jsperf.com/foreach-vs-for-katherine/1) also shows that it's much slower to use `forEach`. 

This code gave me very similar performance results (~0.2% difference):
```
'use strict';

const testObject = {
  a: 1,
  b: 2,
  c: 3,
  d: 4,
  e: 5,
  f: 6,
  g: 7,
  h: 8,
  i: 9,
  j: 10, k: 11, l: 12, m: 13, n: 14, o: 15, p: 16, q: 17, r: 18, s: 19, t:
20, u: 21, v: 22, w: 23, x: 24, y: 25, z: 26,
  hello: 3,
  why: 17,
  aa: 2,
  ab: 3, ac: 2, ad: 4, ae: 5, af: 6, ag: 7, ah: 2, ai: 2, aj: 2, ak: 2,
  al: 2, am: 2, an: 3, ao: 3, ap: 3, aq: 3, ar: 3, as: 3, at: 3, au: 3, av: 3,
  aw: 3, ax: 3, ay: 3, az: 3
};

let avg = 0;
let result = '';
let i = 0;
const numTrials = 10000;
while (i < numTrials) {
  const hrstart = process.hrtime();
  Object.keys(testObject).forEach(testKey => {
    if (testKey === 'a') {
      console.log('a');
    }
  });
/*
  const testKeys = Object.keys(testObject);
  for (let i = 0, n = testKeys.length; i < n; i++) {
    if (testKeys[i] === 'a') {
      console.log('a');
    }
  }
*/
  const hrend = process.hrtime(hrstart);
  avg += hrend[1];
  i++;
}
const finalAvg = avg / numTrials;
console.log(`avg: ${finalAvg}`);
```

When I changed `console.log('a')` to `result = 'a'`, the `forEach` approach took 2-3 times as long as the `for` loop.

Reply:
That makes sense to me. Since `Object.keys(...).forEach`is very readable I think that we should go with that for now. If we find out that performance is an issue, I can easily change this to use the `for` loop approach.